devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -0,0 +1,634 @@
+	commands << zwave.associationV1.associationSet(groupingIdentifier:6, nodeId:[zwaveHubNodeId])
+	commands << zwave.multiChannelV3.multiChannelEndPointGet()
+	commands + getRefreshCommands()
+
+	encapCommands(commands)
+}
+
+def parse(String description) {
+	log.debug "parse() / description: ${description}"
+
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+
+	log.debug "Parse returned ${result}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	// Preferences template begin
+	log.debug "Configuration report: ${cmd}"
+	def preference = parameterMap.find( {it.parameterNumber == cmd.parameterNumber} )
+	def key = preference.key
+	def preferenceValue = getPreferenceValue(preference, cmd.scaledConfigurationValue)
+
+	if(state.currentPreferencesState."$key".status == "reverseSyncPending"){
+		log.debug "reverseSyncPending"
+		state.currentPreferencesState."$key".value = preferenceValue
+		state.currentPreferencesState."$key".status = "synced"
+	} else {
+		def preferenceKey = preference.key
+		def settingsKey = settings."$key"
+		log.debug "preference.key: ${preferenceKey}"
+		log.debug "settings.key: ${settingsKey}"
+		log.debug "preferenceValue: ${preferenceValue}"
+
+		if (settings."$key" == preferenceValue) {
+			state.currentPreferencesState."$key".value = settings."$key"
+			state.currentPreferencesState."$key".status = "synced"
+		} else if (preference.type == "boolRange") {
+			if (state.currentPreferencesState."$key".status == "disablePending" && preferenceValue == preference.disableValue) {
+				state.currentPreferencesState."$key".status = "disabled"
+			} else {
+				runIn(5, "syncConfiguration", [overwrite: true])
+			}
+		} else {
+			state.currentPreferencesState."$key"?.status = "syncPending"
+			runIn(5, "syncConfiguration", [overwrite: true])
+		}
+	}
+	// Preferences template end
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" ? preference.optionActive : preference.optionInactive
+		case "boolRange":
+			def parameterKeyBoolean = parameterKey + "Boolean"
+			return !notNullCheck(settings."$parameterKeyBoolean") || settings."$parameterKeyBoolean" ? settings."$parameterKey" : preference.disableValue
+		case "range":
+			return settings."$parameterKey"
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private notNullCheck(value) {
+	return value != null
+}
```

Comment:
I don't see the value of breaking this out as a function

Reply:
ok, fixed.